examples/docs_beta_snippets/docs_beta_snippets/guides/data-modeling/partitioning/time_based_partitioning.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+import pandas as pd
+
+import dagster as dg
+
+# Create the PartitionDefinition
+daily_partitions = dg.DailyPartitionsDefinition(start_date="2024-01-01")
+
+
+# Define the partitioned asset
+@dg.asset(partitions_def=daily_partitions)
+def daily_sales_data(context: dg.AssetExecutionContext) -> None:
+    date = context.partition_key
+    # Simulate fetching daily sales data
+    df = pd.DataFrame({"date": [date], "sales": [1000]})
+
+    os.makedirs("daily_sales", exist_ok=True)
+    filename = f"daily_sales/sales_{date}.csv"
+    df.to_csv(filename, index=False)
+
+    context.log.info(f"Daily sales data written to {filename}")
+
+
+@dg.asset(
+    partitions_def=daily_partitions,
+    deps=[daily_sales_data],
+)
+def daily_sales_summary(context):
+    partition_date_str = context.partition_key
+    # Read the CSV file for the given partition date
+    filename = f"daily_sales/sales_{partition_date_str}.csv"
+    df = pd.read_csv(filename)
+
+    # Summarize daily sales
+    summary = {
+        "date": partition_date_str,
+        "total_sales": df["sales"].sum(),
+    }
+
+    context.log.info(f"Daily sales summary for {partition_date_str}: {summary}")
+
+
+# Create a partitioned asset job
+daily_sales_job = dg.define_asset_job(
+    name="daily_sales_job",
+    selection=[daily_sales_data, daily_sales_summary],
+    partitions_def=daily_partitions,
+)
+
+
+# Create a schedule to run the job daily
+@dg.schedule(
+    job=daily_sales_job,
+    cron_schedule="0 1 * * *",  # Run at 1:00 AM every day
+)
+def daily_sales_schedule(context):
+    date = context.scheduled_execution_time.strftime("%Y-%m-%d")
```

Comment:
Should this be offset by -1, I think this would ask for the partition for the current day

Reply:
Darn you are very right
<img width="1253" alt="image" src="https://github.com/user-attachments/assets/9a209fd9-bc44-436a-8ef7-e7f9867abe76">


Docstring is confusing but accurate
> The first partition in the set will start at the start_date at midnight. The last partition
    in the set will end before the current time, unless the end_offset argument is set to a
    positive number.